BuildResidentialHPXML/measure.rb

```diff
@@ -3241,6 +3255,22 @@ def self.create(runner, model, args, epw_file, skip_error_checking)
       end
     end
 
+    if args[:apply_defaults].is_initialized
+      apply_defaults = args[:apply_defaults].get
+    else
+      apply_defaults = false
+    end
+
+    if apply_defaults
+      eri_version = Constants.ERIVersions[-1]
+      epw_path = args[:weather_station_epw_filepath]
+      if not File.exist? epw_path
+        epw_path = File.join(File.expand_path(File.join(File.dirname(__FILE__), '..', 'weather')), epw_path) # a filename was entered for weather_station_epw_filepath
+      end
+      weather, epw_file = Location.apply_weather_file(model, runner, epw_path, '')
```

Comment:
This code seems largely duplicative of [code found elsewhere in the measure](https://github.com/NREL/OpenStudio-HPXML/blob/9cbea592a2f3a07abf75366e084e992604cced7c/BuildResidentialHPXML/measure.rb#L2906-L2914). Can you consolidate these code blocks and then just pass around the, e.g. `epw_file`, as needed?